logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationInteractionAvailabilityUseCase.kt

```diff
@@ -0,0 +1,69 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.fold
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+
+/** TODO
+ * Use case that check if self user is able to interact in conversation
+ * @param conversationId the id of the conversation where user checks his interaction availability
+ * @return an [IsInteractionAvailableResult] containing Success or Failure cases
+ */
+class ObserveConversationInteractionAvailabilityUseCase internal constructor(
+    private val conversationRepository: ConversationRepository,
+    private val selfUserId: UserId,
+) {
+    suspend operator fun invoke(conversationId: ConversationId): Flow<IsInteractionAvailableResult> {
+        return conversationRepository.observeIsUserMember(conversationId, selfUserId)
+            .combine(conversationRepository.observeConversationDetailsById(conversationId)) { eitherIsSelfMember, eitherConversation ->
+                eitherIsSelfMember.fold({ failure -> IsInteractionAvailableResult.Failure(failure) }, { isSelfMember ->
+                    eitherConversation.fold({ failure -> IsInteractionAvailableResult.Failure(failure) }, { conversationDetails ->
```

Comment:
If we don't want to do any special handling on error, shouldn't we use flatmap instead and just return error on the last step?